include/StochasticNoiseSimulator.hpp

```diff
@@ -3,189 +3,142 @@
+        amplitudeDampingProb = (cAmpDampingProbability) ? cAmpDampingProbability.value() : noiseProbability * 2;
+        multiQubitGateFactor = cMultiQubitGateFactor;
 
-    std::map<std::string, double> StochSimulate();
+        if (noiseProbability < 0 || amplitudeDampingProb < 0 || noiseProbability * multiQubitGateFactor > 1 || amplitudeDampingProb * multiQubitGateFactor > 1) {
+            throw std::runtime_error("Error probabilities are faulty!"
+                                     "\n single qubit error probability: " +
+                                     std::to_string(cNoiseProbability) +
+                                     " multi qubit error probability: " + std::to_string(cNoiseProbability * multiQubitGateFactor) +
+                                     "\n single qubit amplitude damping  probability: " + std::to_string(amplitudeDampingProb) +
+                                     " multi qubit amplitude damping  probability: " + std::to_string(amplitudeDampingProb * multiQubitGateFactor));
+        }
 
-    [[nodiscard]] std::size_t getMaxMatrixNodeCount() const override { return 0; } // Not available for stochastic simulation
+        sequentiallyApplyNoise = unoptimizedSim;
+        stochasticRuns         = cStochRuns;
+        setRecordedProperties(recordedProperties);
+        Simulator<DDPackage>::dd->resize(qc->getNqubits());
+        setMaxInstances();
+    }
 
-    [[nodiscard]] std::size_t getMatrixActiveNodeCount() const override { return 0; } // Not available for stochastic simulation
+    std::vector<std::pair<long, std::string>>        recordedProperties;
+    std::vector<std::vector<double>>                 recordedPropertiesPerInstance;
+    std::vector<double>                              finalProperties;
+    std::vector<std::map<std::string, unsigned int>> classicalMeasurementsMaps;
+    std::map<std::string, unsigned int>              finalClassicalMeasurementsMap;
 
-    [[nodiscard]] std::size_t countNodesFromRoot() const override { return 0; } // Not available for stochastic simulation
+    void setRecordedProperties(const std::string& input);
+    void setMaxInstances() { maxInstances = std::thread::hardware_concurrency() > 4 ? std::thread::hardware_concurrency() - 4 : 1; };
+
+    std::map<std::string, std::size_t> Simulate(unsigned int shots) override;
+    std::map<std::string, double>      StochSimulate();
+
+    [[nodiscard]] std::size_t    getMaxMatrixNodeCount() const override { return 0U; }    // Not available for stochastic simulation
+    [[nodiscard]] std::size_t    getMatrixActiveNodeCount() const override { return 0U; } // Not available for stochastic simulation
+    [[nodiscard]] std::size_t    countNodesFromRoot() const override { return 0U; }       // Not available for stochastic simulation
+    [[nodiscard]] dd::QubitCount getNumberOfQubits() const override { return qc->getNqubits(); };
+    [[nodiscard]] std::size_t    getNumberOfOps() const override { return qc->getNops(); };
```

Comment:
why are these defined here again? their function is the same as in the base class.

Reply:
Removed lines